apps/editing-toolkit/editing-toolkit-plugin/whats-new/class-wp-rest-wpcom-whats-new-list.php

```diff
@@ -0,0 +1,78 @@
+<?php
+/**
+ * WP_REST_WPCOM_Whats_New_List file.
+ *
+ * @package A8C\FSE
+ */
+
+namespace A8C\FSE;
+
+use Automattic\Jetpack\Connection\Client;
+
+/**
+ * Class WP_REST_WPCOM_Whats_New_List.
+ */
+class WP_REST_WPCOM_Whats_New_List extends \WP_REST_Controller {
+	/**
+	 * WP_REST_WPCOM_Block_Editor_Whats_New_Dot_Controller constructor.
+	 */
+	public function __construct() {
+		$this->namespace = 'wpcom/v2';
+	}
+
+	/**
+	 * Register available routes.
+	 */
+	public function register_routes() {
+		register_rest_route(
+			$this->namespace,
+			'/whats-new/announcements',
+			array(
+				array(
+					'methods'             => \WP_REST_Server::READABLE,
+					'callback'            => array( $this, 'get_announcements' ),
+					'permission_callback' => array( $this, 'permission_callback' ),
+					'args'                => array(
+						'_locale' => array(
+							'type'        => 'string',
+							'description' => 'locale of the client app, also variations are accepted but filtered out in logic',
+						),
+						'app_id'  => array(
+							'type'        => 'integer',
+							'description' => 'unique application id of the client app (for example getting a 1 here is WORDPRESS_ANDROID_APP_ID)',
+						),
+					),
+				),
+			)
+		);
+	}
+
+	/**
+	 * Get json of announcements to be displayed
+	 *
+	 * @param \WP_REST_Request $request Request object.
+	 * @return mixed
+	 */
+	public function get_announcements( \WP_REST_Request $request ) {
+		$locale   = $request->get_params()['_locale'];
+		$args     = array( 'method' => 'get' );
+		$endpoint = "whats-new/list?_locale={ $locale }";
+		$response = Client::wpcom_json_api_request_as_blog( $endpoint, 'v2', $args, null, 'wpcom' );
```

Comment:
Based on how `wpcom_json_api_request_as_blog` is used here, does it mean an HTTP call will also be made on wpcom? Do I need to create a helper class that access wpcom database directly for the announcements and use that if `is_wpcom` and use `wpcom_json_api_request_as_blog` only for atomic/jetpack sites?

Reply:
I haven't used this `Client::wpcom_json_api_request_as_blog` before, but it appears to be this method in Jetpack, and it does the `is_wpcom` check to save the network request internally so you don't have to.
https://github.com/Automattic/jetpack/blob/master/projects/packages/connection/src/class-client.php#L447

I think I'd check that the `Client::wpcom_json_api_request_as_blog` function exists before calling it. Although it's not supported, it's totally possible for any WordPress site to [install the ETK plugin from the plugin directory](wordpress.org/plugins/full-site-editing/), even if the don't have Jetpack installed. It's not supported so we don't have to worry about it working, but we should at least make sure it doesn't crash. If the function doesn't exist we can just return some error response.

Reply:
Do we return an error that's displayed to the user? eg. 'Jetpack installation is required' or just to catch error to prevent app from crashing?

Reply:
I don't think it matters if the user doesn't see the error, I'm just happy for it not to crash 😄